app/lib/tool/test_profile/importer.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart';
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+
+import 'package:client_data/package_api.dart';
+
+import '../../account/backend.dart';
+import '../../account/models.dart';
+import '../../package/backend.dart';
+import '../../publisher/models.dart';
+import '../../shared/urls.dart' as urls;
+
+import 'models.dart';
+import 'resolver.dart';
+
+/// Imports [profile] data into the Datastore.
+@visibleForTesting
+Future<void> importProfile({
+  @required TestProfile profile,
+  List<String> resolvedVersions,
+  @required String archiveCachePath,
```

Comment:
Can we use `.dart_tool/pub-dev/test_package_cache/` we don't have to hardcode a default here. This is just a suggestion for later :D

Reply:
Yeah, we can do that, but we also want to have a configuration for CI to cache it there too.